src/tiledb/cloud/bioimg/helpers.py

```diff
@@ -34,3 +35,35 @@ def serialize_filter(filter):
         return filter_dict
     else:
         raise TypeError
+
+
+def is_folder(path: str) -> bool:
+    return path.endswith("/")
+
+
+def validate_io_paths(source: Sequence[str], output: Sequence[str]):
+    if not isinstance(source, list) or not isinstance(output, list):
+        raise ValueError("Both source and output must be lists.")
+
+    if len(source) == 0:
+        raise ValueError("Source list must not be empty.")
+
+    # Case 1: If source is a list of files
+    if all(not is_folder(path) for path in source):
+        # Destination is a folder
+        if len(output) == 1 and is_folder(output[0]):
+            return True
```

Comment:
this looks like it either returns True or raises; in that case I would make this `-> None` and use `return` rather than `return True`